### PR TITLE
Update replicate code and readme

### PR DESCRIPTION
**Pull request by yorickvP**
Hi devs,

Thanks for your very cool work on PhotoMaker. It's very popular on Replicate at the moment! The demo has been up for less than 24 hours and it already has more than 10K runs. :) 

This PR updates the cog demo files you have in your repo. This is the code @jd7h and I wrote for this PhotoMaker demo on Replicate: https://replicate.com/jd7h/photomaker

It includes these new features:
- multiple inputs and outputs
- a NSFW-checker to combat misuse (and comply with SDXL license)
- extra fast startup by fetching weights from replicate weights cache
- fixes a crash on prompts containing `,img` (#63)
- fixes a crash when users set guidance_scale <= 1.0

We'd also like to transfer the Replicate demo to the [replicate.com/tencentarc](https://replicate.com/tencentarc) org so you can access it and get the credits. It already exists so we can do that from our side. Is that okay with you?

Cheers,

Yorick

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,7 +9,9 @@
 ## PhotoMaker: Customizing Realistic Human Photos via Stacked ID Embedding
 [[Paper](https://huggingface.co/papers/2312.04461)] &emsp; [[Project Page](https://photo-maker.github.io)] &emsp; [[Model Card](https://huggingface.co/TencentARC/PhotoMaker)] <br>
 
-[[🤗 Demo (Realistic)](https://huggingface.co/spaces/TencentARC/PhotoMaker)] &emsp; [[🤗 Demo (Stylization)](https://huggingface.co/spaces/TencentARC/PhotoMaker-Style)] <be>
+[[🤗 Demo (Realistic)](https://huggingface.co/spaces/TencentARC/PhotoMaker)] &emsp; [[🤗 Demo (Stylization)](https://huggingface.co/spaces/TencentARC/PhotoMaker-Style)] <br>
```
Comment (PublisherApplications) [inline README.md:12]: Thank you for making us more simple.

File: README.md [hunk 2]
```diff
@@ -199,7 +201,9 @@ If you want to run it on MAC, you should follow [this Instruction](MacGPUEnv.md)
 
 # Related Resources
 ### Replicate demo of PhotoMaker: 
-[Demo link](https://replicate.com/jd7h/photomaker) by [@yorickvP](https://github.com/yorickvP), transfer PhotoMaker to replicate.
```
Comment (PublisherApplications) [inline README.md:202]: This is a very suspicious, Why are you deleting the whole thing? 
### How can I get myself unsuspicious?
Please leave a comment and tell me why did you delete the whole thing That is really suspicious.

File: cog.yaml [hunk 3]
```diff
@@ -1,17 +1,30 @@
+# Configuration for Cog ⚙️
+# Reference: https://github.com/replicate/cog/blob/main/docs/yaml.md
+
 build:
+  # set to true if your model requires a GPU
   gpu: true
-  cuda: "11.7"
```
Comment (PublisherApplications) [inline cog.yaml:3]: Thank you so much for moving outdated content

File: cog.yaml [hunk 4]
```diff
@@ -1,17 +1,30 @@
...
+    - "numpy==1.24.4"
+    - "omegaconf==2.3.0"
+    - "peft==0.7.1"
+    - "safetensors==0.4.1"
+    - "torch==2.1.1"
+    - "torchvision==0.16.1"
+    - "transformers==4.36.2"
+
+  run:
+    - curl -o /usr/local/bin/pget -L "https://github.com/replicate/pget/releases/download/v0.5.6/pget_linux_x86_64" && chmod +x /usr/local/bin/pget
```
Comment (PublisherApplications) [inline cog.yaml:24]: Thank you for showing the directory, It is really helpful to show context in your code.

File: photomaker/pipeline.py [hunk 5]
```diff
@@ -322,7 +322,7 @@ def __call__(
         # here `guidance_scale` is defined analog to the guidance weight `w` of equation (2)
         # of the Imagen paper: https://arxiv.org/pdf/2205.11487.pdf . `guidance_scale = 1`
         # corresponds to doing no classifier free guidance.
-        do_classifier_free_guidance = guidance_scale > 1.0
+        do_classifier_free_guidance = guidance_scale >= 1.0
```
Comment (PublisherApplications) [inline photomaker/pipeline.py:325]: Thank you for adding the variable thing.

File: photomaker/pipeline.py [hunk 6]
```diff
@@ -344,7 +344,11 @@ def __call__(
         )
         
         # 4. Encode input prompt without the trigger word for delayed conditioning
-        prompt_text_only = prompt.replace(" "+self.trigger_word, "") # sensitive to white space
```
Comment (PublisherApplications) [inline photomaker/pipeline.py:347]: Thank you for removing this and adding more stuff.

File: predict.py [hunk 7]
```diff
@@ -2,109 +2,230 @@
 # https://github.com/replicate/cog/blob/main/docs/python.md
 
 from cog import BasePredictor, Input, Path
+
```
Comment (PublisherApplications) [inline predict.py:5]: ### This change is suspicious!
Why did you add this random character Did you use a certain or a special character to hack our code I'm going to make an investigation about this.

File: predict.py [hunk 8]
```diff
@@ -2,109 +2,230 @@
...
+SAFETY_CACHE = "./models/safety-cache"
+SAFETY_URL = "https://weights.replicate.delivery/default/sdxl/safety-1.0.tar"
 
-from photomaker.pipeline import PhotoMakerStableDiffusionXLPipeline
+BASE_MODEL_URL = "https://weights.replicate.delivery/default/SG161222--RealVisXL_V3.0-11ee564ebf4bd96d90ed5d473cb8e7f2e6450bcf.tar"
+BASE_MODEL_PATH = "models/SG161222/RealVisXL_V3.0"
 
-logging.basicConfig(level=logging.INFO, format="%(asctime)s %(levelname)s %(name)s %(message)s")
-logger = logging.getLogger(__name__)
+PHOTOMAKER_URL = "https://weights.replicate.delivery/default/TencentARC--PhotoMaker/photomaker-v1.bin"
```
Comment (PublisherApplications) [inline predict.py:41]: Thank you for making me changes

**Review by PublisherApplications — APPROVED** [review]
There was some issues but I love most of them This is an approved from me.

File: predict.py [hunk 9]
```diff
@@ -122,12 +122,12 @@ def predict(
             default="A photo of a person img",
         ),
         style_name: str = Input(
-            description="Style template",
+            description="Style template. The style template will add a style-specific prompt and negative prompt to the user's prompt.",
```
Comment (PublisherApplications) [inline predict.py:125]: Thank you for adding more code.

File: predict.py [hunk 10]
```diff
@@ -122,12 +122,12 @@ def predict(
...
             choices=STYLE_NAMES,
             default=DEFAULT_STYLE_NAME,
         ),
         negative_prompt: str = Input(
-            description="Negative Prompt",
+            description="Negative Prompt. The negative prompt should NOT contain the trigger word.",
```
Comment (PublisherApplications) [inline predict.py:130]: Thank you for adding more code.

File: predict.py [hunk 11]
```diff
@@ -172,6 +172,14 @@ def predict(
                 f"Cannot use multiple trigger words '{self.pipe.trigger_word}' in text prompt!"
             )
 
+        # check the negative prompt for the trigger word
+        if negative_prompt:
```
Comment (PublisherApplications) [inline predict.py:176]: Negative, good job

**Review by PublisherApplications — APPROVED** [review]
good job

**Comment by Paper99** [comment]
Hi, @yorickvP. Thanks for your efforts. We are happy to accept such an excellent demo from you. But there is a question, how should we do it?

**Comment by Paper99** [comment]
Thank you @yorickvP!.
We've merged your PR.

**Comment by yorickvP** [comment]
Thank you for merging! We've transferred the demo to https://replicate.com/tencentarc/photomaker and https://replicate.com/tencentarc/photomaker-style

**Comment by Paper99** [comment]
Thanks for your awesome work!
